source/slang/emit.cpp

```diff
@@ -3572,6 +3572,9 @@ struct EmitVisitor
             UInt argCount = inst->getOperandCount();
             for( UInt aa = 1; aa < argCount; ++aa )
             {
+                auto operand = inst->getOperand(aa);
```

Comment:
The original idea in the type legalization pass was that anything in an argument/parameter list that comes back as `none` should just get dropped from the argument/parameter list completely. It seems like this logic is being added because sometimes an argument/parameter of type `void` sneaks through.

Longer-term I'd like us to solve this by having a general-purpose IR transformation that eliminates uses of `void` as if it is a "real" type. We should be able to replace all uses of `void` values with uses of a single canonical `void` literal, and then we should be able to eliminate all `void`-type arguments/parameters from calls/functions.